pkg/oc/admin/migrate/templateinstances/templateinstances.go

```diff
@@ -0,0 +1,179 @@
+
+		The following transformations will occur:
+
+%s`, prettyPrintMigrations(transforms)))
+
+	internalMigrateTemplateInstancesExample = templates.Examples(`
+		# Perform a dry-run of updating all objects
+	  %[1]s
+
+	  # To actually perform the update, the confirm flag must be appended
+	  %[1]s --confirm`)
+)
+
+func prettyPrintMigrations(versionKinds map[apiType]apiType) string {
+	lines := make([]string, 0, len(versionKinds))
+	for initial, final := range versionKinds {
+		line := fmt.Sprintf("		- %s.%s --> %s.%s", initial.APIVersion, initial.Kind, final.APIVersion, final.Kind)
+		lines = append(lines, line)
+	}
+	sort.Strings(lines)
+
+	return strings.Join(lines, "\n")
+}
+
+type MigrateTemplateInstancesOptions struct {
+	templateClient templateinternalclient.TemplateInterface
+
+	migrate.ResourceOptions
+
+	transforms map[apiType]apiType
+}
+
+// NewCmdMigrateTemplateInstancesAPI implements a MigrateTemplateInstances command
+func NewCmdMigrateTemplateInstances(name, fullName string, f *clientcmd.Factory, in io.Reader, out, errout io.Writer) *cobra.Command {
+	options := &MigrateTemplateInstancesOptions{
+		ResourceOptions: migrate.ResourceOptions{
+			In:     in,
+			Out:    out,
+			ErrOut: errout,
+
+			AllNamespaces: true,
+			Include:       []string{"templateinstance"},
+		},
+		transforms: transforms,
+	}
+	cmd := &cobra.Command{
+		Use:     name,
+		Short:   "Update TemplateInstances to point to the latest group-version-kinds",
+		Long:    internalMigrateTemplateInstancesLong,
+		Example: fmt.Sprintf(internalMigrateTemplateInstancesExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(options.Complete(name, f, cmd, args))
+			kcmdutil.CheckErr(options.Validate())
+			kcmdutil.CheckErr(options.Run())
+		},
+	}
+	options.ResourceOptions.Bind(cmd)
+
+	return cmd
+}
+
+func (o *MigrateTemplateInstancesOptions) Complete(name string, f *clientcmd.Factory, c *cobra.Command, args []string) error {
+	if len(args) != 0 {
+		return fmt.Errorf("%s takes no positional arguments", name)
+	}
+
+	o.ResourceOptions.SaveFn = o.save
+	if err := o.ResourceOptions.Complete(f, c); err != nil {
+		return err
+	}
+
+	clientConfig, err := f.ClientConfig()
+	if err != nil {
+		return err
+	}
+	templateClient, err := templateclientset.NewForConfig(clientConfig)
+	if err != nil {
+		return err
+	}
+	o.templateClient = templateClient.Template()
+
+	return nil
+}
+
+func (o MigrateTemplateInstancesOptions) Validate() error {
+	return o.ResourceOptions.Validate()
+}
+
+func (o MigrateTemplateInstancesOptions) Run() error {
+	return o.ResourceOptions.Visitor().Visit(func(info *resource.Info) (migrate.Reporter, error) {
+		return o.checkAndTransform(info.Object)
+	})
+}
+
+func (o *MigrateTemplateInstancesOptions) checkAndTransform(templateInstanceRaw runtime.Object) (migrate.Reporter, error) {
+	templateInstance, wasTI := templateInstanceRaw.(*templateapi.TemplateInstance)
```

Comment:
optional nit: standard is `ok`, not `wasTI`, right?

Reply:
ditto, copied from legacyhpa.  frankly I like the wasX naming better :)